contrib/olivere/elastic/elastictrace.go

```diff
@@ -115,7 +118,22 @@ func peek(rc io.ReadCloser, max int, n int) (string, io.ReadCloser, error) {
 	}
 	snip, err := r.Peek(n)
 	if err == io.EOF {
-		err = nil
+		return string(snip), rc2, nil
 	}
+
+	if encoding == "gzip" {
+		// unpack the snippet
+		gzr, err := gzip.NewReader(bytes.NewReader(snip))
+		if err != nil {
+			return string(snip), rc2, nil
+		}
+		defer gzr.Close()
+
+		snip, err = ioutil.ReadAll(gzr)
+		if err != nil {
```

Comment:
You can completely remove this condition, it will just fall through to the next line, which does the same.